### PR TITLE
Benchmark build testing

### PR DESCRIPTION
**Pull request by DominicDams**
### Issues:
CryptoAlg-1093

### Description of changes: 
Adds CI dimensions for OpenSSL 1.1, 3.0 and aws-lc benchmarking tools

### Testing:
Built and tested the CI in my test account

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/ci/run_benchmark_build_tests.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+run_build -DOPENSSL_INSTALL_DIR="${install_dir}/openssl-1.1"
+"${BUILD_ROOT}/tool/ossl_bm"
+
+build_openssl_3_0
+echo "Testing ossl_bm with OpenSSL 3.0"
+run_build -DOPENSSL3_INSTALL_DIR="${install_dir}/openssl-3.0"
+"${BUILD_ROOT}/tool/ossl3_bm"
+
+# echo "Testing ossl_bm with OpenSSL 1.1"
+# run_build -DOPENSSL_INSTALL_DIR="${install_dir}"
```
Comment (andrewhop) [inline tests/ci/run_benchmark_build_tests.sh:55]:
```suggestion
```

File: tests/ci/run_benchmark_build_tests.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+source tests/ci/common_posix_setup.sh
+
+install_dir="$(pwd)/../librypto_install_dir"
+openssl_url='https://github.com/openssl/openssl.git'
+openssl_1_1_branch='OpenSSL_1_1_1-stable'
+openssl_3_0_branch='openssl-3.0'
+
+function build_openssl_1_1 {
```
Comment (andrewhop) [inline tests/ci/run_benchmark_build_tests.sh:12]: Indentation is off for both of these functions.

File: tests/ci/run_benchmark_build_tests.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
+#!/bin/bash -ex
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+source tests/ci/common_posix_setup.sh
+
+install_dir="$(pwd)/../librypto_install_dir"
+openssl_url='https://github.com/openssl/openssl.git'
+openssl_1_1_branch='OpenSSL_1_1_1-stable'
+openssl_3_0_branch='openssl-3.0'
```
Comment (andrewhop) [inline tests/ci/run_benchmark_build_tests.sh:10]: Small feature creep: can we add in a build with BoringSSL?

File: tests/ci/run_benchmark_build_tests.sh [hunk 4]
```diff
@@ -0,0 +1,55 @@
+#!/bin/bash -ex
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+source tests/ci/common_posix_setup.sh
+
+install_dir="$(pwd)/../librypto_install_dir"
```
Comment (andrewhop) [inline tests/ci/run_benchmark_build_tests.sh:7]: NP: to improve readability can this SRC_ROOT or BUILD_ROOT?

File: tests/ci/run_benchmark_build_tests.sh [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+openssl_1_1_branch='OpenSSL_1_1_1-stable'
+openssl_3_0_branch='openssl-3.0'
+
+function build_openssl_1_1 {
+echo "building OpenSSL 1.1"
+git clone --branch "${openssl_1_1_branch}" "${openssl_url}" "../openssl-1.1"
+pushd "../openssl-1.1"
+mkdir -p "${install_dir}/openssl-1.1"
+./config --prefix="${install_dir}/openssl-1.1" --openssldir="${install_dir}/openssl-1.1"
+make
```
Comment (andrewhop) [inline tests/ci/run_benchmark_build_tests.sh:18]: To speed this up please specify the number of parallel jobs:
```suggestion
make "-j${NUM_CPU_THREADS}"
```

File: tests/ci/run_benchmark_build_tests.sh (hunk 3 above)
Comment (DominicDams) [reply tests/ci/run_benchmark_build_tests.sh:10]: BoringSSL isn't currently working with our benchmarking tools :(

File: tests/ci/run_benchmark_build_tests.sh (hunk 4 above)
Comment (DominicDams) [reply tests/ci/run_benchmark_build_tests.sh:7]: This is where the OpenSSL build is being installed to

File: tests/ci/run_benchmark_build_tests.sh (hunk 3 above)
Comment (samuel40791765) [reply tests/ci/run_benchmark_build_tests.sh:10]: It's one of the todos to fix within the SIM :)

File: tests/ci/cdk/cdk/codebuild/github_ci_linux_x86_omnibus.yaml [hunk 6]
```diff
@@ -380,6 +380,14 @@ batch:
         compute-type: BUILD_GENERAL1_SMALL
         image: ECR_REPO_PLACEHOLDER:ubuntu-20.04_clang-9x_latest
 
+    - identifier: ubuntu2004_clang9x_x86_64_benchmark
+      buildspec: ./tests/ci/codebuild/linux-x86/run_bm_build_test.yml
+      env:
+        type: LINUX_CONTAINER
+        privileged-mode: true
+        compute-type: BUILD_GENERAL1_SMALL
+        image: ECR_REPO_PLACEHOLDER:ubuntu-20.04_clang-7x-bm-framework_latest
```
Comment (samuel40791765) [inline tests/ci/cdk/cdk/codebuild/github_ci_linux_x86_omnibus.yaml:389]: The compiler version in the "image" and "identifier" don't align

**Review by samuel40791765 — APPROVED** [review]

**Pull request merged**